### ISSUE TITLE
`upper_case_acronyms` does not accept allow lint on enum variants

### ISSUE THREAD

**Issue by Qwaz** [rust-lang/rust-clippy#7708]
I ran Clippy on this code:

```rust
enum TestAttr {
    Test,
    #[allow(clippy::upper_case_acronyms)]
    ATTR,
}
```

I expected to see this happen:
`ATTR` name is allowed

Instead, this happened:
Clippy still emits an error

It happens because `UpperCaseAcronyms` is visiting enum variants manually instead of using `check_variant()` that handles the attributes automatically.

https://github.com/rust-lang/rust-clippy/blob/ef2e2f0a0c61ac29986ac1cee5c06406783a89a2/clippy_lints/src/upper_case_acronyms.rs#L115-L122

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.57.0-nightly (fdf65053e 2021-09-07)
binary: rustc
commit-hash: fdf65053e99e8966f9bd83b5a8491326cb33d638
commit-date: 2021-09-07
host: x86_64-unknown-linux-gnu
release: 1.57.0-nightly
LLVM version: 13.0.0
```

**Comment by Qwaz** [comment]
Can be fixed by swithing to `span_lint_hir()` or `span_lint_hir_and_then()` and providing the field's `HirId`

**Comment by mateusmedeiros** [comment]
Hit the same bug here. Thought I was doing something wrong, haha.

For anyone that might stumble here, the workaround is to allow upper_case_acronyms on the whole Enum. You, of course, are sacrificing that one lint for the whole of your enum, so it's up to you to make sure you're not using uppercase acronyms in any other variant you may add (besides the one that presumably is a false positive you wanted to allow).

```rust
#[allow(clippy::upper_case_acronyms)]
enum TestAttr {
    Test,
    ATTR,
    HTTPResponse // <- Clippy won't catch this one
}
```
